modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js

```diff
@@ -183,6 +174,45 @@ class SegmentEdit extends Component {
 		);
 	}
 
+	/**
+	 * Validates fields with validation and prevents the default form submission
+	 * if there are any errors.
+	 *
+	 * Form submission is defined by the action attribute on the <form> element
+	 * outside of this component. Since we are leveraging the <aui:form> taglib
+	 * to handle submission and formik to handle value changes and validation,
+	 * this method uses formik to validate and prevents the taglib form action
+	 * from being called.
+	 * @param {Class} event Event to prevent a form submission from occurring.
+	 */
+	_handleValidate = event => {
+		const {validateForm} = this.props;
+
+		event.persist();
```

Comment:
What exactly is this being called for?

Reply:
I don't fully understand it, but since event.preventDefault() is being called asynchronously, all event properties will be null, and calling event.persist() retains the original values.

Without calling persist first, attempting to call `event.preventDefault()` throws the error:

> Warning: This synthetic event is reused for performance reasons. If you're seeing this, you're accessing the method `preventDefault` on a released/nullified synthetic event. This is a no-op function. If you must keep the original synthetic event around, use event.persist(). See https://fb.me/react-event-pooling for more information.